### PR TITLE
Run2-TB36 Also save cumulative energy deposit for pasive volumes in the list

### PR DESCRIPTION
**Pull request by bsunanda**
Earlier it was recording the energy deposited in the leaf volume only. Now in addition it record the energy deposited in this volume as well as any volume which is a daughter, grand-daughter, ... of this volume.

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda , can you, please, clarify what is the difference between hit energy and cumulative hit energy?

**Comment by bsunanda** [comment]
@civanch I believe I tried to explain what it does now

**Comment by civanch** [comment]
@bsunanda , this is strange to me, because it is difficult to imagine that a very detailed information for fine structure geometry will be needed. If by construction a separate hits are created without any accumulation then we may have large number of hits, which will take big volume. Potentially, it is possible to before creation of a new hit to check to add it to one of existing (event if they are not in the same volume but are in daughter or granddaughter). This will be a real saving of number of hits. An addition of cumulative energy will not do that. Of course, technically I have no problem with this PR.

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc [hunk 1]
```diff
@@ -79,67 +80,84 @@ void HGCPassive::update(const BeginOfRun * run) {
...
+      } else {
+	time = (aStep->GetPostStepPoint()->GetGlobalTime());
+	copy = (unsigned int)(touchable->GetReplicaNumber(0) + 
+			      1000*touchable->GetReplicaNumber(1));
+      }
+      if (it != mapLV_.end()) {
+	storeInfo(it, plv, copy, time, energy, true);
       } else if (topLV_ != nullptr) {
-	auto itr = findLV(topLV_);
+	auto itr = (init_) ? mapLV_.find(topLV_) : findLV(topLV_);
```
Comment (davidlange6) [inline SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc]: hi @bsunanda - why is this change needed?

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc [hunk 2]
```diff
@@ -79,67 +80,84 @@ void HGCPassive::update(const BeginOfRun * run) {
...
       }
     }//if (curSD==NULL)
+
+    //Now for the mother volumes
+    int level = (touchable->GetHistoryDepth());
+    if (level > 0) {
+      double energy = (aStep->GetTotalEnergyDeposit())/CLHEP::GeV;
+      double time   = (aStep->GetTrack()->GetGlobalTime());
+
+      for (int ii=0; ii<level; ++ii) {
```
Comment (davidlange6) [inline SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc]: how about

for ( int i=level; i>0; i--) {

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc [hunk 3]
```diff
@@ -79,67 +80,84 @@ void HGCPassive::update(const BeginOfRun * run) {
...
+	int i = level - ii;
+	G4LogicalVolume* plv = touchable->GetVolume(i)->GetLogicalVolume();
+	auto it = (init_) ? mapLV_.find(plv) : findLV(plv);
+#ifdef EDM_ML_DEBUG
+	edm::LogVerbatim("ValidHGCal") << "Level: " << ii << ":" << i << " "
+				       << plv->GetName() <<" flag in the List "
+				       << (it != mapLV_.end());
+#endif
+	if (it != mapLV_.end()) {
+	  unsigned int copy = (ii == level) ? 0 :
```
Comment (davidlange6) [inline SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc]: ii can never == level in this loop

**Comment by davidlange6** [comment]
-1
pending answers to comments

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc (hunk 1 above)
Comment (bsunanda) [reply SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc]: Earlier it was in a different if-block. There is no need to do the test. I am putting it back in old sequence

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc (hunk 3 above)
Comment (bsunanda) [reply SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc]: Sorry it should be i == level and not ii == level

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc (hunk 2 above)
Comment (bsunanda) [reply SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch @kpedro88 Can you sign the revised version again?

**Comment by kpedro88** [comment]
+1

**Comment by bsunanda** [comment]
@civanch Please approve

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1
